backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/shipment/pickingterminal/Picking_Terminal_StepDef.java

```diff
@@ -96,4 +112,118 @@ public void pickQty(@NonNull final DataTable dataTable)
 
 		processPickingCandidatesCommand.execute();
 	}
+
+	@And("create M_PickingCandidate for M_HU")
+	public void create_M_PickingCand_for_M_HU(@NonNull final DataTable dataTable)
+	{
+		final List<Map<String, String>> rows = dataTable.asMaps();
+		for (final Map<String, String> row : rows)
+		{
+			final String shipmentScheduleIdentifier = DataTableUtil.extractStringForColumnName(row, I_M_ShipmentSchedule.COLUMNNAME_M_ShipmentSchedule_ID + "." + TABLECOLUMN_IDENTIFIER);
+			final de.metas.inoutcandidate.model.I_M_ShipmentSchedule shipmentSchedule = shipmentScheduleTable.get(shipmentScheduleIdentifier);
+
+			final String huIdentifier = DataTableUtil.extractStringForColumnName(row, I_M_HU.COLUMNNAME_M_HU_ID + "." + TABLECOLUMN_IDENTIFIER);
+			final I_M_HU hu = huTable.get(huIdentifier);
+
+			final BigDecimal qtyPicked = DataTableUtil.extractBigDecimalForColumnName(row, I_M_Picking_Candidate.COLUMNNAME_QtyPicked);
+			final String status = DataTableUtil.extractStringForColumnName(row, I_M_Picking_Candidate.COLUMNNAME_Status);
+			final String pickStatus = DataTableUtil.extractStringForColumnName(row, I_M_Picking_Candidate.COLUMNNAME_PickStatus);
+			final String approvalStatus = DataTableUtil.extractStringForColumnName(row, I_M_Picking_Candidate.COLUMNNAME_ApprovalStatus);
+
+			final I_M_Picking_Candidate pickingCandidate = InterfaceWrapperHelper.newInstance(I_M_Picking_Candidate.class);
+			pickingCandidate.setStatus(status);
+			pickingCandidate.setPickStatus(pickStatus);
+			pickingCandidate.setApprovalStatus(approvalStatus);
+			pickingCandidate.setPickFrom_HU_ID(hu.getM_HU_ID());
+			pickingCandidate.setQtyPicked(qtyPicked);
+			pickingCandidate.setM_ShipmentSchedule_ID(shipmentSchedule.getM_ShipmentSchedule_ID());
+			pickingCandidate.setC_UOM_ID(UomId.EACH.getRepoId());
+
+			saveRecord(pickingCandidate);
+		}
+	}
+
+	@And("process picking")
+	public void process_picking(@NonNull final DataTable dataTable)
+	{
+		final List<Map<String, String>> rows = dataTable.asMaps();
+		for (final Map<String, String> row : rows)
+		{
+			final String huIdentifier = DataTableUtil.extractStringForColumnName(row, COLUMNNAME_M_HU_ID + "." + TABLECOLUMN_IDENTIFIER);
+			final String shipmentScheduleIdentifier = DataTableUtil.extractStringForColumnName(row, COLUMNNAME_M_ShipmentSchedule_ID + "." + TABLECOLUMN_IDENTIFIER);
+
+			final I_M_HU hu = huTable.get(huIdentifier);
+			assertThat(hu).isNotNull();
+
+			final I_M_ShipmentSchedule shipmentSchedule = shipmentScheduleTable.get(shipmentScheduleIdentifier);
+			assertThat(shipmentSchedule).isNotNull();
+
+			final String errorMessage = DataTableUtil.extractStringOrNullForColumnName(row, "OPT.ErrorMessage");
+
+			try
+			{
+				pickingCandidateService.processForHUIds(ImmutableSet.of(HuId.ofRepoId(hu.getM_HU_ID())), ShipmentScheduleId.ofRepoId(shipmentSchedule.getM_ShipmentSchedule_ID()));
```

Comment:
i saw this elsewhere further up. please look into extracting the common code into StepDefUtil or a similar place